File: two-sum.js
var twoSum = function (nums, target) {
  var hasher = {};

  for (i = 0; i <= nums.length; i++) {
    if (hasher[nums[i]] !== undefined) {
      return [hasher[nums[i]], i];
    }

    hasher[target - nums[i]] = i;
  }

  return [];
};

console.log(twoSum([2, 7, 11, 15], 9));
console.log(twoSum([3, 2, 4], 6));
console.log(twoSum([3, 3], 6));

File: twoSum.go
package algorithms

func twoSum(nums []int, target int) []int {
	dict := make(map[int]int)
	for index, item := range nums {
		if val, ok := dict[target-item]; ok {
			return []int{val, index}
		}
		dict[item] = index
	}
	return []int{}
}
